_posts/2022-07-27-osbuild-operator-for-os-lifecycle-in-flotta.markdown

```diff
@@ -0,0 +1,199 @@
+export CLUSTER_DOMAIN='mycluster.example.com'
+oc get secret -n osbuild nexus-osbuild-admin-credentials -o jsonpath={.data.password} | base64 -d | xargs nexus3 login --url https://nexus-osbuild-osbuild.apps.${CLUSTER_DOMAIN} --no-x509_verify --username admin --password
+```
+
+Create a Hosted Raw repository named disk-images
+```shell
+nexus3 repository create hosted raw disk-images
+```
+
+Upload the rhel qcow2 image
+```shell
+nexus3 upload rhel-8.6-x86_64-kvm.qcow2 disk-images
+```
+
+### Create an S3 service
+Deploy MiniO
+```shell
+oc apply -n osbuild -f config/creating_env/deploy_minio.yaml
+
+```
+
+Create a bucket name osbuild-images. You can use the aws cli (please follow the [installation instructions](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html))
+```shell
+export CLUSTER_DOMAIN='mycluster.example.com'
+AWS_ACCESS_KEY_ID=minioadmin AWS_SECRET_ACCESS_KEY=minioadmin aws --endpoint-url https://minio-s3-osbuild.apps.${CLUSTER_DOMAIN} --no-verify-ssl s3 mb s3://osbuild-images
+```
+
+Create a secret for the S3 credentials
+```shell
+oc create secret generic osbuild-s3-credentials -n osbuild --from-literal=access-key-id=minioadmin --from-literal=secret-access-key=minioadmin
+
+```
+Create a secret for the CA Bundle using the OCP route
+```shell
+oc get secrets -n openshift-ingress-operator router-ca -o "jsonpath={.data.tls\.crt}" | base64 -d > /tmp/ca-bundle
+oc create secret generic osbuild-s3-ca-bundle -n osbuild --from-file=/tmp/ca-bundle
+
+```
+
+### Create a Secret for your Red Hat Credentials
+
+Find your Red Hat credentials and create a secret:
+```shell
+oc create secret generic redhat-portal-credentials -n osbuild --from-literal=username=<USERNAME> --from-literal=password=<PASSWORD>
+```
+
+### Create a PSQL Server
+Currently, the controller does not support creating the PSQL server on its own, making this step mandatory
+
+Edit the file `config/creating_env/psql.env` and set a real Password
+
+Create the PSQL server by running 
+```shell
+oc new-app --env-file config/creating_env/psql.env postgresql:13-el8 -n osbuild
+```
+
+Edit the file `config/creating_env/postgress_secret.yaml` with the same Password
+
+Create new secret (please enter a real encoded password) 
+```shell
+oc create -f config/creating_env/postgress_secret.yaml
+```
+
+### Provision an External Worker VM using CNV
+Create an SSH key-pair
+```shell
+ssh-keygen -t rsa -b 4096 -C cloud-user@external-builder -f ~/.ssh/external-builder
+```
+
+Create symlinks to the files to facilitate the next step
+```shell
+ln -s ~/.ssh/external-builder.pub config/creating_env/ssh-publickey
+ln -s ~/.ssh/external-builder config/creating_env/ssh-privatekey
+```
+
+Generate the secret
+```shell
+oc create secret generic external-builder-ssh-pair --from-file=config/creating_env/ssh-privatekey --from-file=config/creating_env/ssh-publickey -n osbuild
+```
+
+Deploy the VM
+```shell
+oc apply -n osbuild -f config/creating_env/external-worker-vm.yaml
```

Comment:
should a step to check the vm is running be added?

Reply:
I added one more step

Reply:
I don't see that you did. In any case it should be in the form of an `oc wait`. That is:
`oc wait --for=condition=Ready vm/external-builder`

Reply:
I thought I pushed it sorry